### ISSUE TITLE
Creation of transaction fails for Assets created from dashboard with "Generate new MockHSM key" option

### ISSUE THREAD

**Issue by smalaichami** [chain/chain#402]
Hi, 

Creation of issuance transaction fails for Assets created from dashboard with "Generate new MockHSM key" option.

Works well for the assets created using dashboard If you provide an existing xpub key. 

#### Environment 
1. Chain Core Version - 	1.0.2
2. Chain Core SDK Language and Version - Ruby  
3. Host (e.g. Mac App, Windows App, Docker) - Windows 8.1

#### What steps will reproduce the problem?
1. Create an asset from dashboard with Generate new MockHSM key option
2. Create an account 
3. Submit issuance transaction using API 

#### What result did you expect?
Transaction successfully created..
#### What result did you observe?
Chain::APIError: Code: CH735 Message: Transaction rejected Detail: validation fa
iled in script execution, input 0 (program [0x7b7d DROP DUP TOALTSTACK SHA3 0x58
7f76b87e2a4151a0331037ecd15cc7a4a76d249ce15669543ab8d5417cc293 0x01 0x01 CHECKMU
LTISIG VERIFY FROMALTSTACK FALSE CHECKPREDICATE] args [ 207d71e944839c0e76cb0c37
839b64c8a252c6252e95647a370a5f51701ae152c0ae87]): VERIFY failed Request-ID: 5af7
4f4ca6f53b5c71b1

**Comment by dominic** [comment]
@smalaichami Thanks for reporting this. 

Can you share the the code you're using that fails to submit the transaction when the key is generated from Dashboard? I'm most interested in the parts where you build and submit the transaction, and where your `HsmSigner` object is initialized. That will help me reproduce the issue.

**Comment by smalaichami** [comment]
@dominic I reproduced the error and captured the steps in powerpoint slides. 
Please go through, and I am attaching the slides, scripts and errorlog for reference. 

[chaincore_tx_error.pptx](https://github.com/chain/chain/files/719312/chaincore_tx_error.pptx)
[create_acct_trans.zip](https://github.com/chain/chain/files/719324/create_acct_trans.zip)
[errorlog.txt](https://github.com/chain/chain/files/719325/errorlog.txt)

**Comment by dominic** [comment]
Thanks - reviewing it now.

**Comment by dominic** [comment]
Hi @smalaichami - in your example `create_acct_trans.rb`, you are not adding the generated key to the signer. In your first script, you create a new key and add it as a signature requirement to the assets you are creating:

```rb
key = chain.mock_hsm.keys.create
signer.add_key(key, chain.mock_hsm.signer_conn)

chain.assets.create(alias: 'gold', root_xpubs: [key.xpub], quorum: 1)
chain.assets.create(alias: 'silver', root_xpubs: [key.xpub], quorum: 1)
...
```

This key is used by the signer to sign transactions for the accounts and assets present. In your second script, you create a new key and add it to the HSM signer, but that key is _not_ a key that can sign for your assets.

When you choose to generate a new key for an asset from the Dashboard, you have two options to sign transactions for that asset programatically:

1. Navigate to http://localhost:1999/dashboard/mockhsms, copy your generated key, and add it as a second key to the signer in your script.
2. In your script, query the MockHsm for a key matching a specific alias that you define. The signature for that can be found [here](https://chain.com/docs/ruby/doc/Chain/MockHSM/Key/ClientModule.html). Here's an example of using it in code:

    ```
    matchingKeys = chain.mock_hsm.keys.query({aliases: ['my-key-alias']}).all
    ```

To summarize, when signing a transaction, the HSM must be provided with all public keys required for the transaction. In this case, your HSM signer object needs to know both keys you have created: the key you've assigned to the account, and the key you've generated for your asset via Dashboard.
